va/validation-authority_test.go

```diff
@@ -258,23 +232,33 @@ func TestSimpleHttpTLS(t *testing.T) {
 }
 
 func TestSimpleHttp(t *testing.T) {
-	va := NewValidationAuthorityImpl(true)
+	va := NewValidationAuthorityImpl(false)
 	va.DNSResolver = &mocks.MockDNS{}
 
 	tls := false
 	chall := core.Challenge{Type: core.ChallengeTypeSimpleHTTP, Token: expectedToken, TLS: &tls}
 
+	// NOTE: We do not attempt to shut down the server. The problem is that the
+	// "wait-long" handler sleeps for ten seconds, but this test finishes in less
+	// than that. So if we try to call hs.Close() at the end of the test, we'll be
+	// closing the test server while a request is still pending. Unfortunately,
+	// there appears to be an issue in httptest that trips Go's race detector when
+	// that happens, failing the test. So instead, we live with leaving the server
+	// around till the process exits.
+	hs := simpleSrv(t, expectedToken, tls)
+
+	port, err := getPort(hs)
+	test.AssertNotError(t, err, "failed to get test server port")
+
+	// Attempt to fail a challenge by telling the VA to connect to a port we are
+	// not listening on.
+	va.simpleHTTPPort = (port + 1) % 65536
```

Comment:
Ah, interesting point. However, aren't system-allocated ports always in the high range, making them unlikely to collide with services? I'd kind of like to punt on this issue since I've already seriously blown my time box fixing this problem.

However, I will change the % 65536 to instead do port - 1 if port == 65536, since wrapping around to port 0 would Do The Wrong Thing.